maven-osgi-package-plugin/src/main/java/at/bestsolution/maven/osgi/pack/ProductPackagePlugin.java

```diff
@@ -67,16 +67,19 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
         Xpp3Dom launcherArgs = new Xpp3Dom("launcherArgs");
         Xpp3Dom programArgs = new Xpp3Dom("programArgs");
+        StringBuilder programArgsVal = new StringBuilder();
         for (String a : product.launcherArgs.programArguments) {
-            programArgs.setValue(a);
+            programArgsVal.append(a + " ");
         }
+        programArgs.setValue(programArgsVal.toString());
         launcherArgs.addChild(programArgs);
 
         Xpp3Dom vmArgs = new Xpp3Dom("vmArgs");
+        StringBuilder vmArgsVal = new StringBuilder();
         for (Entry<Object, Object> e : product.launcherArgs.vmProperties.entrySet()) {
```

Comment:
You could use there `product.launcherArgs.vmProperties.entrySet().stream().map( e ->  "-D" + e.getKey() + "=" + e.getValue()).collect(Collectors.joining(" "))` instead of the for loop

Reply:
Done.